websites/H/Heroku/presence.ts

```diff
@@ -0,0 +1,166 @@
+		clientId: "1014999001801691246",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1e3);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp,
+			largeImageKey: "logo_sqaure",
+		},
+		{ hostname, pathname, search } = window.location;
+	switch (hostname) {
+		case "www.heroku.com": {
+			presenceData.details = "Browsing";
+			if (pathname === "/home") presenceData.state = "Home page";
+			else {
+				presenceData.state = document.title.match(
+					/(.*?)(?=(?: \| Heroku|$))/
+				)[1];
+			}
+			break;
+		}
+		case "blog.heroku.com": {
+			presenceData.details = "Browsing blog posts";
+			if (pathname !== "/") {
+				presenceData.state = document.title.match(
+					/(.*?)(?=(?: \| Heroku|$))/
+				)[1];
+			}
+			break;
+		}
+		case "brand.heroku.com": {
+			presenceData.details = "Browsing";
+			presenceData.state = "Brand";
+			break;
+		}
+		case "dashboard.heroku.com": {
+			presenceData.details = "Viewing Dashboard";
+			if (pathname === "/apps") presenceData.state = "Apps";
+			else if (pathname === "/new-app")
+				presenceData.details = "Creating new app";
+			else if (pathname.startsWith("/account")) {
+				presenceData.details = "Managing account";
+				switch (pathname) {
+					case "/acount": {
+						presenceData.state = "Account settings";
+						break;
+					}
+					case "/account/applications": {
+						presenceData.state = "Application settings";
+						break;
+					}
+					case "/account/billing": {
+						presenceData.state = "Billing settings";
+						break;
+					}
+					// No default
+				}
+			} else if (pathname.startsWith("/apps/")) {
+				if (await presence.getSetting<boolean>("showNames")) {
+					presenceData.details = `Managing app: '${
+						document.title.match(/(.*?)(?=(?: \| Heroku|$))/)[1]
+					}'`;
+				} else presenceData.details = "Managing app";
+
+				const [subpath, subpath2] = pathname.split("/").slice(3);
+				if (subpath) {
+					switch (subpath) {
+						case "activity": {
+							presenceData.state = "Activity";
+							if (subpath2 === "builds")
+								presenceData.state = "Viewing build log";
+							break;
+						}
+						default: {
+							presenceData.state = `${subpath[0].toUpperCase()}${subpath.slice(
+								1
+							)}`;
+						}
+					}
+				} else presenceData.state = "Overview";
+			} else if (pathname === "/provision-addon") {
+				presenceData.details = "Provisioning an addon";
+				presenceData.state = document.querySelector(
+					".new-app-view > div > div > div:nth-of-type(2) > div:nth-of-type(2)"
+				).textContent;
+			}
+			break;
+		}
+		case "data.heroku.com": {
+			presenceData.details = "Browsing Data";
+			if (pathname === "/") presenceData.state = "Looking at datastores";
+			else if (pathname === "/dataclips")
+				presenceData.state = "Looking at dataclips";
+			else if (pathname.startsWith("/datastores/")) {
+				if (await presence.getSetting<boolean>("showNames")) {
```

Comment:
You should store the setting in a variable at the top of the updateData func. You use it multiple time.